File: front-end/angularclient/src/app/model/Mazzo.ts
import { Carta } from './Carta';

export abstract class Mazzo {
    protected listaCarte: Array<Carta>;

    protected getListaCarte(): Array<Carta> {
        return this.listaCarte;
    }

    public isVuoto(): boolean {
        let vuoto: boolean;
        if (this.listaCarte.length === 0) {
            vuoto = true;
        }else{
            vuoto = false;
        }
        return vuoto;
    }

    public dimensione():number{
        return this.listaCarte.length;
    }

}

File: back-end/angularspringbootdemo/src/main/java/com/primas/angularspringbootdemo/entity/Mazzo.java
package com.primas.angularspringbootdemo.entity;

import java.util.ArrayList;

public abstract class Mazzo {
	
	private ArrayList<Carta> listaCarte = new ArrayList<>();


	public ArrayList<Carta> getListaCarte() {
		return listaCarte;
	}

	public boolean isVuoto() {
		return listaCarte.isEmpty();
	}

	@Override
	public String toString() {
		return listaCarte.toString();
	}

	public int dimensione() {
		return listaCarte.size();
	}

}
